tools/release/sync-stable-blocks.js

```diff
@@ -0,0 +1,141 @@
+/* eslint-disable no-console */
+/**
+ * External dependencies
+ */
+const fs = require( 'fs' );
+const path = require( 'path' );
+
+/**
+ * Constants
+ */
+const BLOCK_LIBRARY_SRC = 'node_modules/@wordpress/block-library/src/';
+
+const REPOSITORY_ROOT = path.dirname( path.dirname( __dirname ) );
+
+/**
+ * The main function of this task.
+ *
+ * Refreshes the PHP files referring to stable @wordpress/block-library blocks.
+ */
+function main() {
+	const blocks = getStableBlocksMetadata();
+	const toolWarningMessage = '// This file was autogenerated by tools/release/sync-stable-blocks.js, do not change manually!';
+
+	// wp-includes/blocks/require-blocks.php
+	console.log( 'Refreshing wp-includes/blocks/require-static-blocks.php...' );
+	const staticBlockFolderNames = blocks
+		.filter( ( metadata ) => ! isDynamic( metadata ) )
+		.map( toDirectoryName )
+		// To the block folder name statement:
+		.map( dirname => `	'${ dirname }',` )
+		.join( "\n" );
+
+	fs.writeFileSync(
+		`${ REPOSITORY_ROOT }/src/wp-includes/blocks/require-static-blocks.php`,
+		`<?php
+
+${ toolWarningMessage }
+// Returns folder names for static blocks necessary for core blocks registration.
+return array(
+${ staticBlockFolderNames }
+);
+`,
+	);
+
+	// wp-includes/blocks/require-blocks.php
+	console.log( 'Refreshing wp-includes/blocks/require-dynamic-blocks.php...' );
+	const dynamicBlockFileRequires = blocks
+		.filter( isDynamic )
+		.map( toDirectoryName )
```

Comment:
I'm wondering if we should sort those directory names to ensure it's always sorted the same way. I see some subtle differences sometimes in my testing:

![Uploading Screenshot 2022-07-04 at 14.23.00.png…]()

Reply:
Done with [1ffc973](https://github.com/WordPress/wordpress-develop/pull/2940/commits/1ffc973e461681ad1d94cf30a4b0dd6f423cd314).